src/app/api/bookings/route.ts

```diff
@@ -0,0 +1,64 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { getBookings } from '@/lib/utils/server/booking';
+
+export async function GET(request: NextRequest) {
+  try {
+    const url = new URL(request.url);
+
+    const bookingName = url.searchParams.get('bookingName');
+    const start = url.searchParams.get('start');
+    const end = url.searchParams.get('end');
+    const venue = url.searchParams.get('venue');
+    const organisation = url.searchParams.get('organisation');
```

Comment:
Use `zod` to validate the type and searchParams, casting it to a `Date` or `number` if necessary.

For an example, refer to https://github.com/usdevs/NUSCweb/blob/main/src/lib/schema/booking.ts